src/lib/kclCommands.ts

```diff
@@ -44,20 +52,45 @@ export function kclCommands(
         const sampleCodeUrl = `https://raw.githubusercontent.com/KittyCAD/kcl-samples/main/${encodeURIComponent(
           data.sample.replace(FILE_EXT, '')
         )}/${encodeURIComponent(data.sample)}`
-        fetch(sampleCodeUrl)
-          .then(async (response) => {
-            if (!response.ok) {
-              console.error('Failed to fetch sample code:', response.statusText)
-              return
-            }
-            const code = await response.text()
+        const sampleSettingsFileUrl = `https://raw.githubusercontent.com/KittyCAD/kcl-samples/main/${encodeURIComponent(
+          data.sample.replace(FILE_EXT, '')
+        )}/${PROJECT_SETTINGS_FILE_NAME}`
 
-            return {
-              sampleName: data.sample,
-              code,
-              method: data.method,
+        Promise.all([fetch(sampleCodeUrl), fetch(sampleSettingsFileUrl)])
+          .then(
+            async ([
+              codeResponse,
+              settingsResponse,
+            ]): Promise<OnSubmitProps> => {
+              if (!(codeResponse.ok && settingsResponse.ok)) {
+                console.error(
+                  'Failed to fetch sample code:',
+                  codeResponse.statusText
+                )
+                return Promise.reject(new Error('Failed to fetch sample code'))
+              }
+              const code = await codeResponse.text()
+              const parsedProjectSettings = parseProjectSettings(
+                await settingsResponse.text()
+              )
+              let projectSettingsPayload: ReturnType<
+                typeof projectConfigurationToSettingsPayload
+              > = {}
+              if (!err(parsedProjectSettings)) {
+                projectSettingsPayload = projectConfigurationToSettingsPayload(
+                  parsedProjectSettings
+                )
+              }
+
```

Comment:
Could we report something that there was an error? Easier to debug when something goes wrong